tests/jballerina-unit-test/src/test/resources/test-src/types/typedesc/typedesc_positive.bal

```diff
@@ -198,22 +198,55 @@ function testCustomErrorTypeDescWithoutConstraint() {
 }
 
 type ImmutableIntArray int[] & readonly;
+type ImmutableTypleArray [int, int, int] & readonly;
+type ImmutableJsonArray json[] & readonly;
+type ImmutableTableArray table<Employee> & readonly;
 
 function testTypeDefWithIntersectionTypeDescAsTypedesc() {
+
+    // tests for cloneWithType
     typedesc<anydata> a = ImmutableIntArray;
     (int|string)[] arr = [1, 2, 3];
     anydata|error b = arr.cloneWithType(a);
-    // https://github.com/ballerina-platform/ballerina-lang/issues/28912
-    //assertEquality(true, (typeof b).toString());
-    //assertEquality(true, b is int[]);
-    //assertEquality(true, (<int[]> checkpanic b).isReadOnly());
-    //assertEquality(<int[]> [1, 2, 3], b);
-
+    assertEquality("typedesc int[] & readonly", (typeof b).toString());
+    assertEquality(true, b is int[]);
+    assertEquality(true, (<int[]> checkpanic b).isReadOnly());
+    assertEquality(<int[]> [1, 2, 3], b);
+
+    typedesc<anydata> t = ImmutableTypleArray;
```

Comment:
I guess passing the type name itself will work for `cloneWithType`.
```ballerina
    anydata|error t1 = arr.cloneWithType(ImmutableTypleArray);
```
So we don't need to create separate typedescs. 
Shall we check this?

Reply:
As I check with @MaryamZi We need to test via a variable. These are actually disabled tests due to the above issue. Hence keep the seperate typedescs.